src/vm_pmu.rs

```diff
@@ -373,6 +373,112 @@ impl VmPmuState {
         }
     }
 
+    // Implementation of PMU counter configuration on platforms with Sscdeleg support
+    // TODO: Use code in the actual implementation
+    #[allow(dead_code)]
+    fn configure_matching_counters_internal(
+        &mut self,
+        counter_index: u64,
+        counter_mask: u64,
+        config_flags: PmuCounterConfigFlags,
+        _event_type: PmuEventType,
+        _event_data: u64,
+    ) -> SbiResult<u64> {
+        // TODO: Finalize the logic for using event_type and event_data
+        use PmuCounterState::*;
+        let pmu_info = pmu::PmuInfo::get()?;
+        let counter_mask = pmu_info.filter_counter_mask(counter_index, counter_mask)?;
+        const FIXED_COUNTERS_MASK: u64 = !0x7;
+        let index = if !config_flags.is_skip_match() {
+            let mut bitmask_iter =
+                CounterMaskIter::new(counter_index, counter_mask & FIXED_COUNTERS_MASK);
+            bitmask_iter
+                .find(|i| matches!(self.counter_state[*i], NotConfigured))
+                .map_or_else(|| Err(SbiError::InvalidParam), |i| Ok(i as u64))
+        } else {
+            // If skip_match is set, the counter must already be configured
+            let state = self
+                .counter_state
+                .get(counter_index as usize)
+                .ok_or(SbiError::InvalidParam)?;
+            if matches!(state, Started(_) | Configured(_)) {
+                Ok(counter_index)
+            } else {
+                Err(SbiError::InvalidParam)?
+            }
+        }?;
+
+        if config_flags.is_clear_value() {
+            CSR.scounterselect.set(index);
+            CSR.shpmcounter.set(0);
+        }
+
+        if config_flags.is_auto_start() {
+            let value = CSR.scountinhibit.get() & !(1 << index);
+            CSR.scountinhibit.set(value);
+        }
+
+        Ok(index as u64)
+    }
+
+    // Implementation of PMU counter start on platforms with Sscdeleg support
+    // TODO: Use code in the actual implementation
+    #[allow(dead_code)]
+    fn start_counters_internal(
+        &mut self,
+        counter_index: u64,
+        counter_mask: u64,
+        start_flags: PmuCounterStartFlags,
+        initial_value: u64,
+    ) -> SbiResult<()> {
+        use PmuCounterState::*;
+        let pmu_info = pmu::PmuInfo::get()?;
+        let counter_mask = pmu_info.filter_counter_mask(counter_index, counter_mask)?;
+        const FIXED_COUNTERS_MASK: u64 = !0x7;
+        let bitmask_iter = CounterMaskIter::new(counter_index, counter_mask & FIXED_COUNTERS_MASK);
```

Comment:
smcountern should be used to filter the programmable ones.
Sscountdeleg also applies to fixed ones. That should be covered.